src/librustc_back/target/aarch64_linux_android.rs

```diff
@@ -18,6 +18,7 @@ pub fn target() -> Target {
         arch: "aarch64".to_string(),
         target_os: "android".to_string(),
         target_env: "".to_string(),
+        target_vendor: "linux".to_string(),
```

Comment:
It looks like LLVM normalize `arm-linux-androideabi` to `arm--linux-androideabi` which in theory means that `target_os` should be "linux" and `target_env` should be "androideabi", but we can't really change that now! Regardless though it looks like "unknown" is the right vendor for the android triples.